lib/file-source/src/file_watcher.rs

```diff
@@ -144,24 +149,32 @@ impl FileWatcher {
     /// This function will attempt to read a new line from its file, blocking,
     /// up to some maximum but unspecified amount of time. `read_line` will open
```

Comment:
Does this function still block?

Reply:
It still does multiple blocking filesystems reads, so yes. This comment predates the addition of the sleep.